polyfills/Map/polyfill.js

```diff
@@ -141,6 +141,9 @@
 
 	Map.length = 0;
 
+	Map.prototype.constructor = Map;
```

Comment:
I think the constructor property is added automatically upon construction of an object.
```js
function Test() {} // => undefined
a = new Test // => Test {}
a.constructor // => function Test() {}
```